source/dxgi/dxgi.cpp

```diff
@@ -126,17 +237,23 @@ static void init_reshade_runtime_d3d(T *&swapchain, unsigned int direct3d_versio
 	}
 	else if (direct3d_version == 12)
 	{
-		if (com_ptr<IDXGISwapChain3> swapchain3; SUCCEEDED(swapchain->QueryInterface(&swapchain3)))
+		if (com_ptr<IDXGISwapChain3> swapchain4; SUCCEEDED(swapchain->QueryInterface(&swapchain4)))
 		{
 			const com_ptr<D3D12CommandQueue> &command_queue = reinterpret_cast<const com_ptr<D3D12CommandQueue> &>(device_proxy);
 
-			const auto runtime = std::make_shared<reshade::d3d12::runtime_d3d12>(command_queue->_device->_orig, command_queue->_orig, swapchain3.get());
-			if (!runtime->on_init(desc))
+			const auto runtime = std::make_shared<reshade::d3d12::runtime_d3d12>(command_queue->_device->_orig, command_queue->_orig, swapchain4.get());
+
+			HWND window = desc.OutputWindow;
+
+			if (window == nullptr)
+				window = UWPWindow;
+
+			if (!runtime->on_init(desc, window))
```

Comment:
This can be simplified by writing:
```c++
if (desc.OutputWindow == nullptr)
    desc.OutputWindow = UWPWindow;

const auto runtime = ...
if (!runtime->on_init(desc))
...
```
This is because `runtime_d3d12` will use the `OutputWindow` field and the `desc` structure is writable anyway. Doing it this way avoids the changes necessary to `runtime_d3d12`.

Reply:
OK. Done